### ISSUE TITLE
Can't start epoch 0 after epoch -1

### ISSUE THREAD

**Issue by HaiyangLiu1997** [vita-epfl/openpifpaf#124]
Firstly, thanks for your great work, currently it's the best framework in this field i found, It contains almost everything and I have learned a lot from your code.
Unfortunately, after I run the 
```
time CUDA_VISIBLE_DEVICES=0 python3 -m openpifpaf.train \
  --lr=1e-3 \
  --momentum=0.95 \
  --epochs=75 \
  --lr-decay 60 70 \
  --batch-size=8 \
  --head-quad=1 \
  --headnets pif paf \
  --square-edge=401 \
  --regression-loss=laplace \
  --lambdas 30 2 2 50 3 3 \
  --freeze-base=1
```

I can start the epoch -1 for pretraining, but it occur some warning before epoch 0 and then stop running, like this:
```
INFO:Trainer:{'type': 'train', 'epoch': -1, 'batch': 980, 'n_batches': 1000, 'time': 0.172, 'data_time': 0.115, 'lr': 0.0001, 'loss': 2.672, 'head_losses': [0.012, 0.199, 0.025, 0.009, 0.219, 0.254]}
INFO:Trainer:{'type': 'train', 'epoch': -1, 'batch': 990, 'n_batches': 1000, 'time': 0.183, 'data_time': 0.106, 'lr': 0.0001, 'loss': 4.544, 'head_losses': [0.017, 0.323, 0.068, 0.015, 0.376, 0.461]}
INFO:Trainer:{'type': 'train-epoch', 'epoch': 0, 'loss': 4.2328, 'head_losses': [0.01342, 0.30742, 0.04972, 0.01276, 0.37865, 0.44724], 'time': 266.9}
INFO:Trainer:{'type': 'config', 'field_names': ['pif.c', 'pif.vec1', 'pif.scales1', 'paf.c', 'paf.vec1', 'paf.vec2']}
/home/ikenaga/.local/lib/python3.6/site-packages/torch/optim/lr_scheduler.py:82: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

if I change the parameters and don't use pretraining, it will also occur this warning and can't start the training:
```
filter for keypoint annotations ...
... done.
Images: 8000
INFO:Trainer:{'type': 'config', 'field_names': ['pif.c', 'pif.vec1', 'pif.scales1', 'paf.c', 'paf.vec1', 'paf.vec2']}
/home/ikenaga/.local/lib/python3.6/site-packages/torch/optim/lr_scheduler.py:82: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

I use conda and ubuntu 18.04, and run the env setup command. I can run the pretraining, it's seems not a env problem.
I read the last warning and try to change the order of lr_scheduler.step() in the loop function at trainer.py line 90~98,it doesn't work. and I find the order of lr and optimizer is right in train_batch.

And this is just a warning ,not a error, I'm confused why it stop training without any error information.
there are some problems with my proxy, I can't download the weight file from the url when running  this code, but after test I can download it form wget, so finally I just add a path in some lines at the  net.py like this:
`checkpoint = torch.utils.model_zoo.load_url(RESNET50_MODEL,'/home/openpifpaf-master')`
that's all of the my changes for your source code.

I'm sorry to bother you but currently I have no idea about this problem, could you give some hints when you are free? thank you!!!

**Comment by svenkreiss** [comment]
I have not encountered anything like this before. Are there any other warning or error messages?

**Comment by svenkreiss** [comment]
I have just trained another model with the latest code and had no problem. 

Are you running out of GPU memory?

**Comment by svenkreiss** [comment]
Haven't heard back from you, so I assume this is solved. Feel free to reopen and add information if the problem is still there.

**Comment by HaiyangLiu1997** [comment]
sorry for late reply, I reset the environment and it can work☺️

Sven Kreiss <notifications@github.com>于2019年9月21日 周六16:00写道：

**Comment by svenkreiss** [comment]
Great! Thanks for sharing. All the best for your project.

**Comment by fh520** [comment]
I met the same question, it can well train without using pretrained model, while using a pretrained model, it didn't work, the message is as follows:
time CUDA_VISIBLE_DEVICES=2,3 python3 -m openpifpaf.train \
  --batch-size=16 \
  --loader-workers=8 \
  --basenet=shufflenetv2x1 \
  --checkpoint=shufflenetv2x1_train_log.epoch075 \
  --head-quad=1 \
  --headnets pif paf \
  --lr=1e-1 \
  --momentum=0.9 \
  --epochs=75 \
  --lr-decay 60 70 \
  --lambdas 30 2 2 50 3 3 \
  --no-pretrain \
  --weight-decay=1e-5 \
  --update-batchnorm-runningstatistics \
  --ema=0.03 \

INFO:openpifpaf.optimize:SGD optimizer
INFO:openpifpaf.network.trainer:{'type': 'config', 'field_names': ['pif.c', 'pif.vec1', 'pif.scales1', 'paf.c', 'paf.vec1', 'paf.vec2']}
/home/luban/miniconda3/envs/py3/lib/python3.6/site-packages/torch/optim/lr_scheduler.py:100: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
CUDA_VISIBLE_DEVICES=2,3 python3 -m openpifpaf.train --batch-size=16      pif  6.69s user 5.26s system 42% cpu 27.968 total
